src/components/common/ColonyHome/ColonySubscription/ColonySubscription.tsx

```diff
@@ -33,43 +39,86 @@ const MSG = defineMessages({
 });
 
 const ColonySubscription = () => {
-  const { colony } = useColonyContext();
-  const { colonyAddress } = colony || {};
+  const { colony, canInteractWithColony } = useColonyContext();
+  const {
```

Comment:
Could these hooks be wrapped up into a, say, `useColonySubscription` hook that only returns the methods/data the view needs, e.g. the handleSubscribe/unsubscribe and the loading state?